prof/src/com/oracle/souffleprof/ProgramRun.java

```diff
@@ -522,9 +522,9 @@ public Relation getRelation(String name) {
         for (int i = 0; i < table.length; i++) {
             for (int j = 0; j < table[0].length; j++) {
                 if (table[i][j] == null) continue;
-                if (table[i][j].getClass() == Double.class) {
+                if (table[i][j] instanceof Double) {
                     new_table[i][j] = formatTime((Double)table[i][j]);
-                } else if (table[i][j].getClass() == Long.class) {
+                } else if (table[i][j] instanceof Long) {
                     new_table[i][j] = formatNum(precision, (Long)table[i][j]);
                 } else {
                     new_table[i][j] = table[i][j];
```

Comment:
It seems that the result on this end is always a string, and the result is only ever used as such, so I've changed it to be a string[][] and added an explicit toString. I'm not sure if erroring out is the correct behaviour here, since it's likely given the quality of the code that we're going to have unexpected stuff come through, and terminating is probably less useful than just printing our best effort and continuing.